File: dsa-interview-crash-course/1426_counting_elements.cpp
#include <unordered_map>
#include <vector>

class Solution {
   public:
    int countElements(std::vector<int> &arr) {
        std::unordered_map<int, int> map;
        for (auto &a : arr) {
            ++map[a];
        }

        int elem = 0;
        for (auto &[key, freq] : map) {
            if (map.find(key + 1) != map.end()) elem += freq;
        }

        return elem;
    }
};

File: 1426_counting_elements.py
from typing import List


class Solution:
    def countElements(self, arr: List[int]) -> int:
        map = {}

        for a in arr:
            if a in map:
                map[a] += 1
            else:
                map[a] = 1

        elem = 0
        for (val, freq) in map.items():
            if (val + 1) in map.keys():
                elem += freq

        return elem